### ISSUE TITLE
AssignIn returns wrong type

### ISSUE THREAD

**Issue by HaimWix** [wix-incubator/carmi#259]
type SomeObject = {
	someField: string
}
function changeSomeField(carmiObject: Graph<SomeObject, F>) {
	carmiObject.assignIn([{someField: {extraProp: true}}])
	carmiObject.get('someField') --TYPE--> Graph<string & {extraProp: boolean}>
}

There are some [examples](https://stackoverflow.com/questions/49682569/typescript-merge-object-types) online for a better typing though it's not complete as well

**Comment by nirnatan** [comment]
I can't understand the usecase? Do you have an example that I can run or that includes the types so I can copy it and see the error?

**Comment by HaimWix** [comment]
Yes, it happens when you want to merge two objects that has different types for same fields.
The use case I had was for replacing breakpoint field
{breakpoint : string, ...rest}
to
{breakpoint: {range: {min:number, max:number}, mediaQuery} ...rest

I no longer intend to write this code, but I think it's a valid case though not common
